Assignment4/NatalyMMR-e22a304/task06.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+"""Task06.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1_OcZbW0lCjVpVSz1kiYvw8Dm72cwkhhW
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example5.rdf", format="xml")
+
+"""Create a new class named Researcher"""
+
+#TO DO
+ns = Namespace("http://somewhere#")
+g.add((ns.Researcher, RDF.type, RDFS.Class))
+
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.1: Create a new class named "University"**"""
+
+#TO DO
+u = Namespace("http://somewhere#")
+g.add((u.University, RDF.type, RDFS.Class))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.2: Add "Researcher" as a subclass of "Person"**"""
+
+#TO DO
+r = Namespace("http://somewhere#")
+g.add((r.Researcher, RDFS.subClassOf, r.Person))
+
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.3: Create a new individual of Researcher named "Jane Smith"**"""
+
+#TO DO
+r = Namespace("http://somewhere#")
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+g.add((r.JaneSmith, VCARD.type, r.Researcher))
+
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.4: Add to the individual JaneSmith the fullName, given and family names**"""
+
+#TO DO
+r = Namespace("http://somewhere#")
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+g.add((r.JaneSmith, VCARD.FN, Literal("Jane Smith")))
+g.add((r.JaneSmith, VCARD.Given, Literal("Jane")))
+g.add((r.JaneSmith, VCARD.Family, Literal("Smith")))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+#TO DO
+r = Namespace("http://somewhere#")
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+g.add((r.UPM, RDF.type, r.University))
+g.add((r.JohnSmith, VCARD.Work, r.UPM))
```

Comment:
Work does not exist in vcard. Create a new property in your ns

Reply:
Done!